cli/command/cli_test.go

```diff
@@ -247,3 +252,31 @@ func TestGetClientWithPassword(t *testing.T) {
 		})
 	}
 }
+
+func TestNewDockerCliAndOperators(t *testing.T) {
+	// Test default operations and also overriding default ones
+	cli, err := NewDockerCli(
+		WithContentTrust(true),
+		WithContainerizedClient(func(string) (clitypes.ContainerizedClient, error) { return nil, nil }),
+	)
+	assert.NilError(t, err)
+
+	assert.Check(t, cli.In() != nil)
+	assert.Check(t, cli.Out() != nil)
+	assert.Check(t, cli.Err() != nil)
+	assert.Equal(t, cli.ContentTrustEnabled(), true)
+	client, err := cli.NewContainerizedEngineClient("")
+	assert.NilError(t, err)
+	assert.Equal(t, client, nil)
+
+	// Apply can modify a dockerCli after construction
```

Comment:
I'd be happier that this test was really checking that if it was asserting that the in/out streams had `FD() != 0` before this apply.

Oh, except it can be 0 for `In` anyway, since that is the `stdin` fd too. Not sure how best to get around that -- perhaps by calling something (like `runVersion`?) to actually output something into the buf and check it went where you hoped it would?

Reply:
If fixed that directly reading or writing to the streams and check with their underlying buffers. 👍

Reply:
Looks good, thanks.